data_taking_scripts/motor.py

```diff
@@ -57,18 +78,61 @@ def top_plate(self):
+        ''' Returns the status of all 3 motors as a list. '''
         curved_mirror_status = self.curved_mirror.get_status()
         bottom_plate_status = self.bottom_plate.get_status()
         top_plate_status = self.top_plate.get_status()
         return [curved_mirror_status, bottom_plate_status, top_plate_status]
 
     def wait_for_motors(self):
+        ''' Waits for all three motors to stop moving and ready to
+            accept commands. '''
         while (get_motor_status() != ['R','R','R']):
             print(get_motor_status())
             time.sleep(1)
         print('done waiting')
 
     def move_to_zero(self):
+        ''' Moves all three motors to 0. '''
         self.curved_mirror.move_to_zero()
         self.bottom_plate.move_to_zero()
         self.top_plate.move_to_zero()
+
+    def move_by_increment(self, increment_distance, dielectric_plate_thickness,
+                          cavity_length_tracker, num_plates):
+        curved_mirror_steps = curved_mirror_distance_to_steps(increment_distance)
+        print(F'Moving curved mirror motor by {curved_mirror_steps} steps')
+        self.curved_mirror.move_steps(curved_mirror_steps)
+        #adjusting bottom dielectric plate
+        cavity_length_tracker = cavity_length_tracker + increment_distance
+        new_plate_separation = plate_separation(cavity_length_tracker,num_plates)
+        diff = initial_plate_separation +increment_distance
+        move_bottom_plate = diff - new_plate_separation
+        bottom_plate_steps = plates_distance_to_steps(move_bottom_plate,dielectric_plate_thickness)
+        print(F'Moving bottom plate motor by {bottom_plate_steps} steps')
+        self.bottom_plate.move_steps(bottom_plate_steps)
+        #adjusting top dielectric plate
+        move_top_plate = new_plate_separation - initial_plate_separation
+        top_plate_steps = plates_distance_to_steps(move_top_plate,dielectric_plate_thickness)
+        print(F'Moving top plate motor by {top_plate_steps} steps')
+        self.top_plate.move_steps(top_plate_steps)
+        return cavity_length_tracker, new_plate_separation
+
+    def plate_separation(self, length, num_plates):
+        return length/(num_plates+1)
+
+    def plates_distance_to_steps(distance,plate_thickness,holder_thickness = (1/4),
+                                 lip_thickness = (1/20),pitch=(1/20),
+                                 steps_per_rotation = 20000):
+        holder_center = holder_thickness/2
+        plate_center = lip_thickness + plate_thickness/2
+        gap = holder_center - plate_center
+        actual_distance = distance + gap
+        num_pitch_lengths = actual_distance/pitch #these many complete rotations
+        steps = steps_per_rotation * num_pitch_lengths
+        return int(round(steps))
+
+    def curved_mirror_distance_to_steps(distance, pitch = (1/20), steps_per_rotation = 20000):
+        num_pitch_lengths = distance/pitch #these many complete rotations
```

Comment:
@parashar2399 After fixing some class function calls, I finally got to this error. I'm not sure what is happening, so I leave it to you to fix it. My suggestion is that once you fix it, you try to take a modemap where you just move the resonator motor. 

```
root@clerk-dripline-python-deployment-7c8cff556c-trscb:/usr/local/src/dripline-python-plugin/data_taking_scripts# python testing_class.py                                                                                                                                     
fail!
fail!
Enter initial resonator length (in cm): 16
Enter the distance to move in cm (Empty cavity modemap is usually 3): 3
Enter the number of measurements needed: 10
done waiting
Starting modemap measurement
Setting na_measurement_status to start_measurement
Logging list of endpoints
Setting na_measurement_status to stop_measurement
Traceback (most recent call last):
  File "testing_class.py", line 42, in <module>
    num_plates)
  File "/usr/local/src/dripline-python-plugin/data_taking_scripts/motor.py", line 108, in move_by_increment
    curved_mirror_steps = self.curved_mirror_distance_to_steps(increment_distance)
  File "/usr/local/src/dripline-python-plugin/data_taking_scripts/motor.py", line 147, in curved_mirror_distance_to_steps
    num_pitch_lengths = distance/pitch #these many complete rotations
TypeError: unsupported operand type(s) for /: 'OrpheusMotors' and 'float'

```